ml/src/main/scala/org/apache/spark/ml/h2o/algos/H2OAlgorithm.scala

```diff
@@ -58,41 +48,41 @@ abstract class H2OAlgorithm[P <: Model.Parameters : ClassTag, M <: SparkModel[M]
     }
 
     val cols = getFeaturesCols().map(col) ++ Array(col(getLabelCol()))
-    val input = hc.asH2OFrame(dataset.select(cols: _*).toDF())
+    val input = H2OContext.getOrCreate(SparkSession.builder().getOrCreate()).asH2OFrame(dataset.select(cols: _*).toDF())
```

Comment:
WDYT about a dedicated field for ```H2OContext```? as it's in the implementation of ```AutoML``` and ```GridSearch```?

Example:
```protected lazy val hc = H2OContext.getOrCreate(SparkSession.builder().getOrCreate())```